MuMuPiKPAT/test/sanjay/selector/psiPrimePiK_MC.C

```diff
@@ -5872,7 +5905,225 @@ void psiPrimePiK_MC::Terminate()
 	}
       }
 
+    TNtupleD* dataNTuple;
+    TString dataTreeName;
+    rooKeysError = kFALSE;
+
+    if (MC) {
+      dataTreeName = "AA_recoVars";
+      dataNTuple = (TNtupleD*)AlexisOut->Get("AA_recoVars");
+    } else {
+        dataNTuple = (TNtupleD*)AlexisOut->Get("hardcuts_variables_bkg");
+        dataTreeName = "hardcuts_variables_bkg";
+      }
+    if ( (dataNTuple=(TNtupleD*)AlexisOut->Get(dataTreeName)) ) {
+      if (dataNTuple->GetEntries() < 1e5) {
+        rooKeysFlag = kTRUE;
+      } else {
+          rooKeysFlag = kFALSE;
+          rooKeysError = kTRUE;
+          cout << "Eff/Bkg fit with RooKeys: input dataset is too large" << endl;
+        }
+    } else {
+        rooKeysFlag = kFALSE;
+        cout << "Eff/Bkg fit with RooKeys: input dataset not found" << endl;
+    }
 
+    if (rooKeysFlag) {
+
+      RooRealVar massKPi("massKPi", "massKPi", 0.5, 2.5);
+      RooRealVar massPsiPi("massPsiPi", "massPsiPi", 3.0, 5.0);
+      RooRealVar cosMuMu("cosMuMu", "cosMuMu", -1., 1.);
+      RooRealVar phi("phi", "phi", -3.15, 3.15);
+
+      double nbins = 100;
+      Int_t nEntries;
+      int nReco = 1;
+      double xx, yy;
+
+      if (MC) {
+
+        RooDataSet data_reco("data_reco", "data_reco", RooArgSet(massKPi, massPsiPi, cosMuMu, phi));
+        RooDataSet data_gen("data_gen", "data_gen", RooArgSet(massKPi, massPsiPi, cosMuMu, phi));
+        dataTreeName = "AA_recoVars";
+        Double_t obs1, obs2, obs3, obs4, obs5;
+        if ( (dataNTuple=(TNtupleD*)AlexisOut->Get(dataTreeName)) ) {
+          dataNTuple->SetBranchAddress("massKPi", &obs1);
+          dataNTuple->SetBranchAddress("massMuMuPi", &obs2);
+          dataNTuple->SetBranchAddress("cosMuMu", &obs3);
+          dataNTuple->SetBranchAddress("phi", &obs4);
+          dataNTuple->SetBranchAddress("B0beauty", &obs5);
+          nEntries = dataNTuple->GetEntries();
+          nReco = nEntries;
+          for (Int_t i=0; i < nEntries; i++) {
+            dataNTuple->GetEntry(i);
+            massKPi = obs1;
```

Comment:
This should not compile...
Anyway, use a loop over the RooRealVar vector and assign the corresponding obs element from the obs vector